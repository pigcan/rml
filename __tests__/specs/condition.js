
const MLTransformer = require('../../src/Transformer');

describe('conditional render', () => {
  it('support simple', (done) => {
    new MLTransformer([
      '<div>',
      '<div r:if="{{l>1}}">',
      '</div>',
      '</div>',
    ].join('\n')).transform((err, code) => {
      expect(code).toMatchSnapshot();
      done();
    });
  });

  it('support ifs', (done) => {
    new MLTransformer([
      '<div>',
      '<div r:if="{{l>1}}">',
      '</div>',
      '<div r:if="{{t>1}}">',
      '</div>',
      '</div>',
    ].join('\n')).transform((err, code) => {
      expect(code).toMatchSnapshot();
      done();
    });
  });

  it('support simple nested', (done) => {
    new MLTransformer([
      '<div>',
      '<div r:if="{{l>1}}">',
      '<template is="x" r:if="{{t>1}}" />',
      '</div>',
      '</div>',
    ].join('\n')).transform((err, code) => {
      expect(code).toMatchSnapshot();
      done();
    });
  });

  it('support full nested', (done) => {
    new MLTransformer([
      '<div>',
      '<div r:if="{{l>1}}">',
      '</div>',
      '<div r:elif="{{l>2}}">',
      '</div>',
      '<div r:else="{{l>3}}">',
      '</div>' +
      '<div></div>',
      '</div>',
    ].join('\n')).transform((err, code) => {
      expect(code).toMatchSnapshot();
      done();
    });
  });

  it('support empty else', (done) => {
    new MLTransformer([
      '<div>',
      '<div r:if="{{l>1}}">',
      '</div>',
      '<div r:else>',
      '</div>',
      '</div>',
    ].join('\n')).transform((err, code) => {
      expect(code).toMatchSnapshot();
      done();
    });
  });

  it('support multiple roots', (done) => {
    new MLTransformer(`
<block r:if="a">1</block>
<view/>
`.trim())
      .transform((err, code) => {
        expect(code).toMatchSnapshot();
        done();
      });
  });

  it('skip comment', (done) => {
    new MLTransformer(`
<!--1-->
<block r:if="a">1</block>
<!--1-->
<block r:elif="a">1</block>
<!--1-->
<view r:else>1</view>
`.trim())
      .transform((err, code) => {
        expect(code).toMatchSnapshot();
        done();
      });
  });
});
