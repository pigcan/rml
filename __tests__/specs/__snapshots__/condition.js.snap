// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`conditional render skip comment 1`] = `
"import React from 'react';

export default function render(data) {
  return (
    (
    (\`a\`) ?
    (
    \`1\`
    )
    :
    (
    (\`a\`) ?
    (
    \`1\`
    )
    :
    (
    (
    <view>
      {\`1\`}
    </view>
    )
    )
    )
    )
  );
};"
`;

exports[`conditional render support empty else 1`] = `
"import React from 'react';

export default function render(data) {
  return (
    <div>
      {
      (
      ((data.l > 1)) ?
      (
      <div>
      </div>
      )
      :
      (
      (
      <div>
      </div>
      )
      )
      )
      }
    </div>
  );
};"
`;

exports[`conditional render support full nested 1`] = `
"import React from 'react';

export default function render(data) {
  return (
    <div>
      {
      (
      ((data.l > 1)) ?
      (
      <div>
      </div>
      )
      :
      (
      ((data.l > 2)) ?
      (
      <div>
      </div>
      )
      :
      (
      ((data.l > 3)) ?
      (
      <div>
      </div>
      )
      :
      null
      )
      )
      )
      }
      <div>
      </div>
    </div>
  );
};"
`;

exports[`conditional render support ifs 1`] = `
"import React from 'react';

export default function render(data) {
  return (
    <div>
      {
      (
      ((data.l > 1)) ?
      (
      <div>
      </div>
      )
      :
      null
      )
      }
      {
      (
      ((data.t > 1)) ?
      (
      <div>
      </div>
      )
      :
      null
      )
      }
    </div>
  );
};"
`;

exports[`conditional render support multiple roots 1`] = `
"import React from 'react';

export default function render(data) {
  return (
    [
    (
    (\`a\`) ?
    (
    \`1\`
    )
    :
    null
    )
    ,
    <view>
    </view>
    ,
    ]
  );
};"
`;

exports[`conditional render support simple 1`] = `
"import React from 'react';

export default function render(data) {
  return (
    <div>
      {
      (
      ((data.l > 1)) ?
      (
      <div>
      </div>
      )
      :
      null
      )
      }
    </div>
  );
};"
`;

exports[`conditional render support simple nested 1`] = `
"import React from 'react';

export default function render(data) {
  return (
    <div>
      {
      (
      ((data.l > 1)) ?
      (
      <div>
        {
        (
        ((data.t > 1)) ?
        (
        $templates[\`x\`].call(this, undefined)
        )
        :
        null
        )
        }
      </div>
      )
      :
      null
      )
      }
    </div>
  );
};"
`;
